### PR TITLE
Supports for `algolia/algoliasearch-client-php` v4

### PR DESCRIPTION
**Pull request by 3bd-ulrahman**

### CODE REVIEW

**Comment by taylorotwell** [comment]
Tests are failing.

File: src/EngineManager.php [hunk 1]
```diff
@@ -39,7 +41,8 @@ public function createAlgoliaDriver()
     {
         $this->ensureAlgoliaClientIsInstalled();
 
-        UserAgent::addCustomUserAgent('Laravel Scout', Scout::VERSION);
+        // UserAgent::addCustomUserAgent('Laravel Scout', Scout::VERSION);
```
Comment (crynobone) [inline src/EngineManager.php]:
```suggestion
```

File: src/EngineManager.php [hunk 2]
```diff
@@ -61,7 +64,8 @@ public function createAlgoliaDriver()
         }
 
         if (is_int($batchSize = config('scout.algolia.batch_size'))) {
-            $config->setBatchSize($batchSize);
+            // $config->setBatchSize($batchSize);
```
Comment (crynobone) [inline src/EngineManager.php]:
```suggestion
```

**Comment by crynobone** [comment]
Fixed tests

File: src/EngineManager.php [hunk 3]
```diff
@@ -61,7 +62,7 @@ public function createAlgoliaDriver()
         }
 
         if (is_int($batchSize = config('scout.algolia.batch_size'))) {
-            $config->setBatchSize($batchSize);
+            (new Event())->setBatchSize($batchSize);
```
Comment (taylorotwell) [inline src/EngineManager.php]: Does this batch size actually persist and get used by the client anywhere?

Comment (crynobone) [reply src/EngineManager.php]: Changed to add `batchSize` directly to `SearchConfig`.

**Pull request merged**